### ISSUE TITLE
Flaky test: TestTermLog.test_output

### ISSUE THREAD

**Issue by mhils** [mitmproxy/mitmproxy#3725]
See  https://github.com/mitmproxy/mitmproxy/pull/3724/checks?check_run_id=316184660#step:6:197:

```
=================================== FAILURES ===================================
__________ TestTermLog.test_output[None-expected_out0-expected_err0] ___________

self = <test.mitmproxy.addons.test_termlog.TestTermLog object at 0x7fa289713c10>
outfile = None, expected_out = ['one', 'three'], expected_err = ['four']
capfd = <_pytest.capture.CaptureFixture object at 0x7fa289713f10>

    @pytest.mark.usefixtures('capfd')
    @pytest.mark.parametrize('outfile, expected_out, expected_err', [
        (None, ['one', 'three'], ['four']),
        (sys.stdout, ['one', 'three', 'four'], []),
        (sys.stderr, [], ['one', 'three', 'four']),
    ])
    def test_output(self, outfile, expected_out, expected_err, capfd):
        t = termlog.TermLog(outfile=outfile)
        with taddons.context(t) as tctx:
            tctx.options.termlog_verbosity = "info"
            tctx.configure(t)
            t.log(log.LogEntry("one", "info"))
            t.log(log.LogEntry("two", "debug"))
            t.log(log.LogEntry("three", "warn"))
            t.log(log.LogEntry("four", "error"))
        out, err = capfd.readouterr()
        assert out.strip().splitlines() == expected_out
>       assert err.strip().splitlines() == expected_err
E       assert ['Traceback (..._call__', ...] == ['four']
E         At index 0 diff: 'Traceback (most recent call last):' != 'four'
E         Left contains 29 more items, first extra item: '  File "/home/runner/work/mitmproxy/mitmproxy/mitmproxy/proxy/server.py", line 121, in handle'
E         Use -v to get the full diff

test/mitmproxy/addons/test_termlog.py:27: AssertionError
```

**Comment by meetmangukiya** [comment]
Ran tests locally, can't reproduce. The CI logs have expired as well. Probably fixed?

**Comment by mrf345** [comment]
Wasn't able to reproduce it locally too. But when i ran the whole test suite with `pytest-repeat` i got couple of  failures `95 failed, 28936 passed, 154 warnings in 1976.71s (0:32:56)` i ran it with `pytest --count=20`.

**Comment by mhils** [comment]
Probably fixed by #4136
